File: src/dataset/javascript/matrix_exercism-ThomasZumsteg/claude/ClaudeSonnet4_matrix_v3.js
var Matrix = function(matrix) {
    const rowStrings = matrix.split("\n");
    const numRows = rowStrings.length;
    const numCols = rowStrings[0].split(" ").length;
    
    this.rows = new Array(numRows);
    for (let i = 0; i < numRows; i++) {
        const rowElements = rowStrings[i].split(" ");
        this.rows[i] = new Array(numCols);
        for (let j = 0; j < numCols; j++) {
            this.rows[i][j] = parseInt(rowElements[j], 10);
        }
    }

    this.columns = new Array(numCols);
    for (let colNum = 0; colNum < numCols; colNum++) {
        this.columns[colNum] = new Array(numRows);
        for (let rowNum = 0; rowNum < numRows; rowNum++) {
            this.columns[colNum][rowNum] = this.rows[rowNum][colNum];
        }
    }
};

module.exports = Matrix;

File: src/dataset/python/matrix_exercism-ThomasZumsteg/claude/ClaudeSonnet4_matrix_v3.py
"""A 2 dimentional matrix object"""

class Matrix(object):
    """Creates a 2d matrix object with accessable rows and columns"""

    def __init__(self, matrix):
        """Copy of the input data"""
        self.matrix = matrix
        self._rows = None
        self._columns = None

    @property
    def rows(self):
        """Matrix displayed as rows, coumns"""
        if self._rows is None:
            self._rows = [[int(n) for n in row.split()]
                          for row in self.matrix.split("\n") if row.strip()]
        return self._rows

    @property
    def columns(self):
        """Matrix displayed as columns, rows"""
        if self._columns is None:
            rows = self.rows
            if rows:
                self._columns = [[rows[j][i] for j in range(len(rows))]
                                for i in range(len(rows[0]))]
            else:
                self._columns = []
        return self._columns